benches/benches/bevy_ecs/scheduling/schedule.rs

```diff
@@ -143,3 +156,109 @@ pub fn empty_schedule_run(criterion: &mut Criterion) {
     });
     group.finish();
 }
+
+fn base_system(mut query: Query<FilteredEntityMut>) {
+    for filtered_entity in &mut query {
+        // we calculate Faulhaber's formula (https://en.wikipedia.org/wiki/Faulhaber%27s_formula) mod 256
+        // with n = value and p = exponent for each entity.
+        // The time is takes to compute this is dependant on the number of entities in the query and
+        // the values in each entity. This is to ensure that the running times between systems are varied.
+        let mut total: Wrapping<u8> = Wrapping(0);
+        let mut exponent: u32 = 1;
+        for component_id in filtered_entity.access().component_reads_and_writes().0 {
+            // find the value of the component
+            let ptr = filtered_entity.get_by_id(component_id).unwrap();
+            // SAFETY: All components have a u8 layout.
+            let value: u8 = unsafe { *ptr.deref::<u8>() };
+            for i in 0..=value {
+                let mut product = Wrapping(1);
+                for _ in 1..=exponent {
+                    product *= Wrapping(i);
+                }
+                total += product;
+            }
+            exponent += 1;
+        }
+
+        // we assign this value to all the components we can write to
+        for component_id in filtered_entity.access().component_reads_and_writes().0 {
+            let ptr = filtered_entity.get_by_id(component_id).unwrap();
+            if filtered_entity.access().has_component_write(component_id) {
+                // SAFETY:
+                // We have exclusive access so the pointer is unique
+                // All components have a u8 layout
+                unsafe {
+                    let value = ptr.assert_unique().deref_mut::<u8>();
```

Comment:
I think you can combine the `get_by_id()` call with the access check if you use `get_mut_by_id()`.  

```suggestion
            if let Some(ptr) = filtered_entity.get_mut_by_id(component_id) {
                // SAFETY:
                // All components have a u8 layout
                unsafe {
                    let value = ptr.with_type::<u8>();
```